app/models/order.rb

```diff
@@ -57,6 +57,8 @@ def set_subtotal
         end
 
         def generate_order_number
-            self.order_number = SecureRandom.hex(10) + Time.now.to_i.to_s
+            if self.order_number.nil?
```

Comment:
This is not bad, but you can do a cool trick on one line here:
`self.order_number ||= SecureRandom.hex(10) + Time.now.to_i.to_s`

Try to find out what `||=` is used for :)